Block/BlockContextManager.php

```diff
@@ -214,4 +212,31 @@ protected function setDefaultExtraCacheKeys(BlockContextInterface $blockContext,
             $blockContext->setSetting('extra_cache_keys', $extraCacheKeys);
         }
     }
+
+    private function resolve(BlockInterface $block, $settings)
+    {
+        $optionsResolver = new OptionsResolver();
+
+        $this->configureSettings($optionsResolver, $block);
+
+        $service = $this->blockService->get($block);
+        $service->setDefaultSettings($optionsResolver, $block);
+
+        $reflector = new \ReflectionMethod($service, 'setDefaultSettings');
```

Comment:
The reflectionCache cache attribute can either be compute once at with a CompilerPass or at runtime.